tests/runtime/test_dump_services.py

```diff
@@ -58,6 +58,33 @@ def test_dump_grpc_services_dir_does_not_exist():
         shutil.rmtree(fake_dir)
 
 
+@pytest.mark.skipif(
+    PROTOBUF_VERSION < 4 and ARM_ARCH, reason="protobuf 3 serialization bug"
+)
+def test_dump_grpc_services_consolidated():
+    with tempfile.TemporaryDirectory() as workdir:
+        dump_grpc_services(workdir, False, consolidate=True)
+        assert os.path.exists(workdir)
+        # Make sure the file names match the expected names for caikit plus
+        # sample_lib
+        # NOTE: Dumping services dumps _all_ data model objects, so we cannot
+        #   do an exact check due to the global descriptor pool and other tests
+        #   that modify it.
+        dumped_files = os.listdir(workdir)
+        assert all(
+            fname in dumped_files
+            for fname in {
+                "caikit_runtime_SampleLib.proto",
+                "caikit_runtime_info.proto",
+                "caikit_runtime_training.proto",
+                "caikit_data_model_common.proto",
+                "caikit_data_model_common_runtime.proto",
+                "caikit_data_model_runtime.proto",
+                "caikit_data_model_sample_lib.proto",
```

Comment:
Should we do some checks to see that at least one of these files contains all of the expected data models?

Reply:
Done